packetbeat/protos/amqp/amqp.go

```diff
@@ -0,0 +1,883 @@
+package amqp
+
+import (
+	"encoding/binary"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/packetbeat/config"
+	"github.com/elastic/beats/packetbeat/procs"
+	"github.com/elastic/beats/packetbeat/protos"
+	"github.com/elastic/beats/packetbeat/protos/tcp"
+	"github.com/elastic/beats/packetbeat/publish"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Amqp struct {
+	Ports                       []int
+	Send_request                bool
+	Send_response               bool
+	Max_body_length             int
+	Parse_headers               bool
+	Parse_arguments             bool
+	Hide_connection_information bool
+	transactions                *common.Cache
+	transactionTimeout          time.Duration
```

Comment:
some minor/possible performance optimization (?):

Some modules store transactions within the TCP stream + do not create intermediate transaction objects adding support for pipelining at the same time. E.g. redis module:

- https://github.com/elastic/beats/blob/master/packetbeat/protos/redis/redis.go#L18
- https://github.com/elastic/beats/blob/master/packetbeat/protos/redis/redis.go#L228
- https://github.com/elastic/beats/blob/master/packetbeat/protos/redis/redis.go#L244